addons/web/static/tests/views/fields/radio_field.test.js

```diff
@@ -261,3 +268,86 @@ test("radio field is empty", async () => {
     expect(".o_radio_input:disabled").toHaveCount(3);
     expect(".o_radio_input:checked").toHaveCount(0);
 });
+
+test("radio selection field with filter, empty list", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <field name="color" widget="radio_selection_with_filter"
+                       context="{'allowed_selection': []}" />
+            </form>
+        `
+    })
+
+    expect(".o_radio_input").toHaveCount(0);
+});
+
+test("radio selection field with filter, single choice", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <field name="color" widget="radio_selection_with_filter"
+                       context="{'allowed_selection': ['black']}" />
+            </form>
+        `
+    })
+
+    expect(".o_radio_input").toHaveCount(1);
+    expect(".o_field_widget").toHaveText("Black");
+});
+
+test("radio selection field with filter, all choices", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <field name="color" widget="radio_selection_with_filter"
+                       context="{'allowed_selection': ['black', 'red']}" />
+            </form>
+        `
+    })
+
+    expect(".o_radio_input").toHaveCount(2);
+    expect(".o_field_widget").toHaveText("Red\nBlack");
+});
+
+test("radio selection field with filter, sync with other field", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 2,
+        arch: /* xml */ `
+            <form edit="0">
+                <group>
+                    <field name="mood" widget="radio" />
+                    <field name="color" widget="radio_selection_with_filter"
+                           context="{'allowed_selection':
+                                ['red']
+                                + (['black'] if mood == 'sad' else [])
+                           }" />
+                </group>
+            </form>
+        `
+    })
+
+    // click on Red and Happy => Black should be hidden
+    click(queryFirst("[name='mood'] input"));
+    click(queryFirst("[name='color'] input"));
+    await animationFrame();
+    expect(".o_radio_input").toHaveCount(3);
```

Comment:
Why 3? Shouldn't it be 1 and then 2?

Is it because it's selecting the other "mood" field? Maybe we can have a more explicit selector.